### PR TITLE
perf: Optimize strip_files function

### PR DESCRIPTION
**Pull request by frankdavid**
NODE-1454

We'd previously run `rm` for each nested file in directories. When a directory contains many files, this gets slow. By running `rm` only once, we can save 10-20s per build.

The output should still be the same, I verified it by taking the checksum of the generated images.

### CODE REVIEW

File: toolchains/sysimage/build_ext4_image.py [hunk 1]
```diff
@@ -91,10 +92,16 @@ def strip_files(fs_basedir, fakeroot_statefile, strip_paths):
...
+                flattened_paths.append(del_path)
         else:
-            subprocess.run(["fakeroot", "-s", fakeroot_statefile, "-i", fakeroot_statefile, "rm", "-rf", target_path])
+            flattened_paths.append(target_path)
 
+    if not flattened_paths:
+        return
+
+    subprocess.run(
+        ["fakeroot", "-s", fakeroot_statefile, "-i", fakeroot_statefile, "rm", "-rf"] + flattened_paths,
```
Comment (raymondk) [inline toolchains/sysimage/build_ext4_image.py:103]: is there a limit to how many arguments the rm command will take?

Comment (frankdavid) [reply toolchains/sysimage/build_ext4_image.py]: I added batching to avoid too many arguments.

**Review by Bownairo — APPROVED** [review]

**Comment by Bownairo** [comment]
> The output should still be the same, I verified it by taking the checksum of the generated images.

I'm curious, at what point were you checking the hash? We have an `echo-component-files-hash` target to check on components without waiting for the full image to build. Maybe we can have more of these at various stages?

**Comment by frankdavid** [comment]
I checked the end result. Let's keep in mind to add some reporting of diffs at certain stages if we do more refactors in the future.

**Pull request merged**